bin/cask

```diff
@@ -324,10 +330,21 @@ def main():
             exit_error(
                 'Python 2.6 required, yours is {0.major}.{0.minor}'.format(
                     sys.version_info))
+
+
         if len(sys.argv) > 1 and sys.argv[1] == 'exec':
+            ## never returns!
             exec_command(sys.argv[2:])
-        else:
-            exec_cask(sys.argv[1:])
+
+        ## special handling for --emacs/bin as this has to happen before
+        ## everything else, and we don't want the parsing in two places.
+        if len(sys.argv) > 1:
+            if sys.argv[1] == '--emacs':
+                global EMACS
```

Comment:
> I think you are making things unnecessarily difficult -- setting at initialization in this case really just means "can be initialized in a single line". 

Yes, and that's pretty much how I'd like to keep it.

> I understand why you don't want to pass changing global state around, but the "never changed once initialized" semantics seems a reasonable distinction to me.

I would normally agree, but see, Python does not actually enforce these kind of distinctions, and I think that they are too subtle to rely purely on programmer's discipline for their enforcement.  My point of view is probably distorted by having used languages with strong type systems, but…

> Still, your code base, so I will change it that way, once I have fixed the lisp problem.

… this pretty much nails it.  In the absence of a strong type systems that helps me to catch mistakes, I'd rather prefer to have very strict rules about how variables should be used.